File: app/src/main/java/com/theandroiddev/mywins/injection/AppComponent.java
package com.theandroiddev.mywins.injection;

import com.theandroiddev.mywins.data.db.AppDatabase;
import com.theandroiddev.mywins.editsuccess.EditSuccessActivity;
import com.theandroiddev.mywins.editsuccess.EditSuccessPresenter;
import com.theandroiddev.mywins.successes.SuccessesActivity;
import com.theandroiddev.mywins.successes.SuccessesPresenter;
import com.theandroiddev.mywins.successslider.SuccessSliderActivity;
import com.theandroiddev.mywins.successslider.SuccessSliderContract;
import com.theandroiddev.mywins.successslider.SuccessSliderPresenter;
import com.theandroiddev.mywins.utils.SuccessesConfig;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by jakub on 07.11.17.
 */

@Singleton
@Component(modules = {AppModule.class, PresenterModule.class})
public interface AppComponent {
    void inject(SuccessesActivity target);

    void inject(SuccessesPresenter target);

    void inject(SuccessSliderActivity target);

    void inject(SuccessSliderPresenter target);

    void inject(SuccessSliderContract.SuccessImageLoader target);

    void inject(EditSuccessActivity target);

    void inject(EditSuccessPresenter target);

    void inject(AppDatabase target);

    void inject(SuccessesConfig target);



}

File: app/src/main/java/com/theandroiddev/mywins/injection/AppComponent.kt
package com.theandroiddev.mywins.injection

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.DaggerApplication
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Created by jakub on 07.11.17.
 */

@Singleton
@Component(modules = [
    AndroidSupportInjectionModule::class,
    ActivityBuildersModule::class,
    AppModule::class,
    RepositoryModule::class,
    CommonModule::class,
    FragmentBuildersModule::class])

interface AppComponent : AndroidInjector<DaggerApplication> {

    override fun inject(application: DaggerApplication)

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent

    }

}

